applications/conceptual-model-editor/src/app/diagram/dialog/select-classes-dialog.tsx

```diff
@@ -0,0 +1,117 @@
+import { useCallback, useState } from "react";
+import { useModelGraphContext } from "../context/model-context";
+import { isSemanticModelClass } from "@dataspecer/core-v2/semantic-model/concepts";
+import { useBaseDialog } from "../components/base-dialog";
+import { useConfigurationContext } from "../context/configuration-context";
+import { CreateButton } from "../components/dialog/buttons/create-button";
+import { CancelButton } from "../components/dialog/buttons/cancel-button";
+import { t } from "../application/";
+import { isSemanticModelClassUsage } from "@dataspecer/core-v2/semantic-model/usage/concepts";
+import { type EntityDetailSupportedType, EntityProxy } from "../util/detail-utils";
+import { tailwindColorToHex } from "~/app/utils/color-utils";
+
+const getDefaultColor = () => {
+    return "#069420";
+};
+
+export const useSelectClassesDialog = () => {
+    const { isOpen, open, close, BaseDialog } = useBaseDialog();
+    const [onSelectConfirmCallback, setOnSelectConfirmCallback] = useState<((newEntityID: string[]) => void) | undefined>(undefined);
+
+    const localOpen = (onSelectConfirmCallback?: (newEntityID: string[]) => void) => {
+        setOnSelectConfirmCallback(onSelectConfirmCallback);
+        open();
+    };
+
+    // TODO: Maybe could be optimized later, when I get to know React better
+    function createChangedRecord<T>(record: Record<string, T>, id: string, newValue: T): Record<string, T> {
+        const newRecord = {...record};
+        newRecord[id] = newValue;
+        return newRecord;
+    }
+
+
+    const SelectClassesDialog = () => {   
+        const [connectionsMap, setConnectionsMap] = useState<Record<string, boolean>>({});
+
+        const { models, aggregatorView } = useModelGraphContext();
+        const activeVisualModel = aggregatorView.getActiveVisualModel();     
+
+        const { language: preferredLanguage } = useConfigurationContext();
+
+        const getPreferredName = useCallback((entity: EntityDetailSupportedType) => {
+            const { name } = EntityProxy(entity, preferredLanguage);
+            return name;
+        }, [preferredLanguage]);
+            
+        
+        // TODO: Probably could use better dependecies, but if it works it works
+        const getSelectedClasses = useCallback(() => {
+            const selectedClasses = Object.entries(connectionsMap).map(([k, v]) => {                
+                if(v) {
```

Comment:
Please do not use single character as a property name. Also be explicit with conditions. Using `if(v)` it is not clear what you test for: not null, not undefined, not false, ..... I'm not sure about the intention here, but better use if (v !== null) ..

Or even better do not use negation and change the code to
```ks
if (v === null) {
  continue;
}
return null.
```

Or you can replace this by use of filter or find functions, something like:
```js
Object.entries(connectionsMap).filter(([key, value]) => value !== null).map(([key, value]) => value);
```